File: components/permissions/src/main/java/org/cloudfoundry/credhub/audit/entities/DeletePermissions.java
package org.cloudfoundry.credhub.audit.entities;

import org.cloudfoundry.credhub.audit.OperationDeviceAction;
import org.cloudfoundry.credhub.audit.RequestDetails;

public class DeletePermissions implements RequestDetails {

  private String credentialName;
  private String actor;

  public DeletePermissions() {
    super();

  }
  public DeletePermissions(final String credentialName, final String actor) {
    super();
    this.credentialName = credentialName;
    this.actor = actor;
  }

  public String getCredentialName() {
    return credentialName;
  }

  public void setCredentialName(final String credentialName) {
    this.credentialName = credentialName;
  }

  public String getActor() {
    return actor;
  }

  public void setActor(final String actor) {
    this.actor = actor;
  }

  @Override
  public OperationDeviceAction operation() {
    return OperationDeviceAction.DELETE_PERMISSIONS;
  }
}

File: components/permissions/src/main/kotlin/org/cloudfoundry/credhub/audit/entities/DeletePermissions.kt
package org.cloudfoundry.credhub.audit.entities

import org.cloudfoundry.credhub.audit.OperationDeviceAction
import org.cloudfoundry.credhub.audit.RequestDetails

class DeletePermissions(
    credentialName: String,
    actor: String,
) : RequestDetails {
    override fun operation(): OperationDeviceAction = OperationDeviceAction.DELETE_PERMISSIONS
}
